File: Algorithm/Leetcode/0000_0100/0011_0020/0020.cpp
// 1720. Decode XORed Array

class Solution {
public:
    vector<int> decode(vector<int>& encoded, int first) {
    	encoded.insert(encoded.begin(), first);
		for (size_t i = 1; i < encoded.size(); i++)
		{
			encoded[i] ^= encoded[i - 1];
		}
		
		return encoded;
        
    }
};

File: Algorithm/Leetcode/0000_0100/0011_0020/0020.py
# 1720. Decode XORed Array

"""https://leetcode.com/problems/decode-xored-array/discuss/1075067/Python-1-Liner-(List-Comprehension-with-Assignment-Expresion)"""

class Solution:
    def decode(self, encoded: List[int], first: int) -> List[int]:
        return [first] + [first:= first ^ x for x in encoded]